src/repr/src/row.rs

```diff
@@ -1468,20 +1403,6 @@ pub fn datum_size_deterministic<const VAR: bool>(datum: &Datum) -> usize {
     }
 }
 
-/// Number of bytes required by a sequence of datums.
-///
-/// This method can be used to right-size the allocation for a `Row`
-/// before calling [`RowPacker::extend`].
-pub fn datums_size_deterministic<'a, I, D, const VAR: bool>(iter: I) -> usize
-where
-    I: IntoIterator<Item = D>,
-    D: Borrow<Datum<'a>>,
-{
-    iter.into_iter()
-        .map(|d| datum_size_deterministic::<VAR>(d.borrow()))
-        .sum()
-}
-
```

Comment:
We might need something like this for arrangement specialization, but we can just get it back from here if we need to.